pkg/logql/quantile_over_time_sketch.go

```diff
@@ -0,0 +1,413 @@
+package logql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql"
+	promql_parser "github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/grafana/loki/pkg/iter"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql/sketch"
+	"github.com/grafana/loki/pkg/logqlmodel"
+)
+
+const (
+	QuantileSketchMatrixType = "QuantileSketchMatrix"
```

Comment:
Nit for this file in general, what do you think about calling things here `ProbabilisticQuantileX`, like:

```
type QuantileSketchVector []quantileSketchSample
type QuantileSketchMatrix []QuantileSketchVector
```
becomes
```
type ProbabilisticQuantileVector []probabilisticQuantilSample
type ProbabilisticQuantileMatrix []probabilisticQuantileVector
```

The actual sketch is in the sketch sub package.

Reply:
Yes! Except that I'm bad a typing this out 😂